saviSimulation/savi/jason_processing/behaviour/UxVBehavior.java

```diff
@@ -0,0 +1,234 @@
+                double azimuth = polar.get(Geometry.AZIMUTH);
+                double elevation = polar.get(Geometry.ELEVATION);
+                double dist = polar.get(Geometry.DISTANCE);
+                if ((azimuth < Math.PI / 2. || azimuth > 3 * Math.PI / 2.) && (dist < perceptionDistance)) {
+                    //it's visible
+                    detectedItems.add(new CameraPerception(wo.getType(), this.time, azimuth, elevation, dist, wo.getPixels() / 2));
+                    visibleItems.add(new CameraPerception(wo.getType(), this.time, azimuth, elevation, dist, wo.getPixels() / 2));
+                }
+            }
+        }
+
+        //remove objects covered by others on the visualization
+        for (CameraPerception di : detectedItems) {
+            for (int i = 0; i < visibleItems.size(); i++) {
+                //to calculate visual angle covered by the object
+                distance = di.getParameters().get(2);
+                //angle deviation from centroid = radius
+                oposite = di.getParameters().get(3);
+                //math to calculate angle cover
+                tan = oposite / distance;
+                angle = Math.abs(Math.atan(tan));
+
+                //if object is covered by di remove
+                //if is covered by azimuth angle
+                if (di.getParameters().get(0) + angle > visibleItems.get(i).getParameters().get(0) && di.getParameters().get(0) - angle < visibleItems.get(i).getParameters().get(0)) {
+                    //if it is covered by elevation angle
+                    if (di.getParameters().get(1) + angle > visibleItems.get(i).getParameters().get(1) && di.getParameters().get(1) - angle < visibleItems.get(i).getParameters().get(1)) {
+                        //if it's at a mayor distance
+                        if (di.getParameters().get(2) < visibleItems.get(i).getParameters().get(2)) {
+                            visibleItems.remove(visibleItems.get(i));
+                        }
+                    }
+                }
+            }
+        }
+
+        return visibleItems;
+    }
+
+    /**
+     * Adds to the current compass angle and normalizes the value between 0 and 2 * PI
+     * @param val
+     */
+    protected void addNormalizedCompassAngle(double val) {
+        this.compasAngle += val;
+        double twoPi = 2 * Math.PI;
+
+        if (this.compasAngle < 0)
+            this.compasAngle += twoPi;
+        if (this.compasAngle > twoPi)
+            this.compasAngle -= twoPi;
+    }
+
+    /**
+     * Creates the default actions for a UxV agent
+     */
+    protected void createActionMap() {
+        actionMap.put("turn(left)", (() -> this.addNormalizedCompassAngle(-Math.PI / 16.0)));
+        actionMap.put("turn(right)", (() -> this.addNormalizedCompassAngle(Math.PI / 16.0)));
+        actionMap.put("thrust(on)", (() -> this.speedVal = SPEED));
+        actionMap.put("thrust(off)", (() -> this.speedVal = 0));
+    }
```

Comment:
Thanks for the heads up! 

Depending on how we update the actions in #31, I can adjust and clean this up to only be coupled with the action names (i.e. "turn") and can pass the action parameters to the lambda. Once the logic for an action becomes more complicated than just a one-liner, we can just extract that AgentAction implementation to it's own class rather than in a lambda!

It shouldn't be too difficult, but I will follow up with the simulation side of things to make sure there aren't any major concerns.